src/test/unit_tests/test_notifications.js

```diff
@@ -171,16 +189,53 @@ mocha.describe('notifications', function() {
             //there shouldn't be a notification for the delete, wait 2 seconds to validate this
             await notify_await_result({timeout: 2000});
         });
+
+        mocha.it('multipart', async () => {
+            const res_create = await s3.createMultipartUpload({
+                Bucket: bucket,
+                Key: 'mp1'
+            });
+
+            const res_upload = await s3.uploadPart({
+                Bucket: bucket,
+                Key: 'mp1',
+                UploadId: res_create.UploadId,
+                PartNumber: 1,
+                Body: 'this is the body'
+            });
+
+            const res_complete = await s3.completeMultipartUpload({
+                Bucket: bucket,
+                Key: 'mp1',
+                UploadId: res_create.UploadId,
+                MultipartUpload: {
+                    Parts: [{
+                        ETag: res_upload.ETag,
+                        PartNumber: 1
+                    }]
+                }
+            });
+
+            await notify_await_result({
+                bucket_name: bucket,
+                event_name: 'ObjectCreated:CompleteMultipartUpload',
+                key: "mp1",
+                etag: res_complete.ETag
```

Comment:
res_complete.Etag and res_upload.Etag should be same.
Do we need to put some assert condition here before notify result?

Reply:
No, that's not correct.
res_upload.Etag is md5(part body), similar to put_object's etag.
res_complete.Etag is md5 of concatenated part's etags.
AWS docs don't explain this explicitly, [this](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html) only has - 
"This ETag is not necessarily an MD5 hash of the object data."